java/code/src/com/redhat/rhn/domain/channel/ChannelFactory.java

```diff
@@ -1450,19 +1450,8 @@ public static void analyzeServerNeededCache() {
      * @param to  the target Channel
      */
     public static void cloneModulesMetadata(Channel from, Channel to) {
-        if (!from.isModular()) {
-            if (to.isModular()) {
-                HibernateFactory.getSession().delete(to.getModules());
-                to.setModules(null);
-            }
-        }
-        else {
-            if (!to.isModular()) {
-                Modules modules = new Modules();
-                modules.setChannel(to);
-                to.setModules(modules);
-            }
-            to.getModules().setRelativeFilename(from.getModules().getRelativeFilename());
-        }
+        to.getModules().clear();
+        from.getModules().forEach(fromModule ->
```

Comment:
I think the from.getModules() can return null. Should it be protected with a isModular?
```suggestion
        if (fromisModular()) {
            from.getModules().forEach(fromModule ->
            ....
        }
```

Reply:
This set is managed by Hibernate and there's no path that sets these null currently, but you're right, we can foolproof it with a check :+1: